### DISCUSSION TITLE
GitHub settings

### DISCUSSION THREAD

**Discussion by danlangford** [pappde/bmai#23] [Ideas]
so i have a PR out for github to do some source code scanning. i like using tools like that to help ensure my code is of high quality

I also noticed that you merged a PR but the source branch was left around. you may consider some of the following settings on your repo

![image](https://user-images.githubusercontent.com/721364/110072698-666e9e00-7d3b-11eb-9151-bdbcdf750d08.png)

I like auto-deleting the branch after the PR was merged.

i also personally like SQUASHING all the commits in the PR into a single commit as it gets merged onto the main branch. that's very much preference though. strong arguments for all those options. many people don't like that but I find it nice to have just a single commit on the main branch that represents the work of the PR. it makes rolling back and identifying where a problem lies easier in my opinion. now in the screen shot I have disallowed merge and rebase as at work we are quite strict. 

feel free to use this thread to discuss, ask questions, debate, etc.

**Comment by pappde** [comment]
I lean towards enabling "squash merge" but leaving the original branch. That way there is some additional context if you find a problem with a particular merge, and need to go back to see more detail. But it keeps the main log of commits clean if you don't need the extra detail.

Of course, the author can always delete the original branch.

**Reply by pappde** [reply]
SQUASH
It seems to be already doing that by default.

DELETE BRANCH
I noticed that github conveniently gives you a "delete branch" button after doing a pull request, so it is behaving pretty well as is.  

However, I selected to "delete branch" for the last couple of PR and while they no longer show up in the github branch list, they still show up when I do a fetch locally of the origin.
